sql/analyzer/resolve_columns.go

```diff
@@ -520,32 +555,85 @@ func resolveColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel
 			return n, transform.SameTree, nil
 		}
 
-		if _, ok := n.(sql.Expressioner); !ok {
-			return n, transform.SameTree, nil
+		var jtNew, left sql.Node
+		var jtSame transform.TreeIdentity
+		var jtErr error
+		var isJoin transform.TreeIdentity
+		switch j := n.(type) {
+		case *plan.CrossJoin:
```

Comment:
Can simplify this with JoinNode, right?

Reply:
Only `InnerJoin`, `LeftJoin`, and `RightJoin` implement `sql.JoinNode` interface